index.js

```diff
@@ -224,20 +210,33 @@ class Analytics {
 
     debug('flush: %o', data)
 
-    request
-      .post(`${this.host}/v1/batch`)
-      .auth(this.writeKey, '')
-      .timeout(this.timeout)
-      .retry(this.retryCount)
-      .send(data)
-      .end((err, res) => {
-        err = err || error(res)
+    const done = err => {
+      callbacks.forEach(callback => callback(err))
+      callback(err, data)
 
-        callbacks.forEach(callback => callback(err))
-        callback(err, data)
+      debug('flushed: %o', data)
+    }
+
+    const req = {
+      method: 'POST',
+      url: `${this.host}/v1/batch`,
+      data
+    }
+
+    if (this.timeout) {
+      req.timeout = typeof this.timeout === 'string' ? ms(this.timeout) : this.timeout
+    }
+
+    axios(req)
```

Comment:
This could be rewritten as:

```js
axios(req)
  .catch(err => {
    if (err.response) {
      err = new Error(err.response.statusText)
    }
    
    return err
  })
  .then(err => {
    callbacks.forEach(callback => callback(err))
    callback(err, data)
  })
```

Reply:
Not quite. The furfilled callback receives the "response" from axios (see https://github.com/axios/axios#response-schema). To get this to work, we could do something like:

```
.catch(err => {
  if (err.response) {
    err = new Error(err.response.statusText)
  }
  return err
}).then(err => {
  if (!(err instanceof Error)) {
    err = undefined
  }
  
  callbacks.forEach(callback => callback(err))
  callback(err, data)

  debug('flushed: %o', data)
})
```

But that feels pretty awkward to me.

Reply:
Hmm, could also do:

```js
axios(req)
.then(res => {}) // ignore the response
.catch(err => {
  if (err.response) {
    err = new Error(err.response.statusText)
  }
  return err
})
.then(err => {
  callbacks.forEach(callback => callback(err))
  callback(err, data)

  debug('flushed: %o', data)
})
```

but this also feels a bit strange.

Reply:
You are right, let's leave it like it is in the PR, with a few minor adjustments (will comment separately).